src/de/tuebingen/uni/ub/hc/MARCProcessing/MARC4JProcessor.java

```diff
@@ -0,0 +1,169 @@
+package de.tuebingen.uni.ub.hc.MARCProcessing;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.TreeSet;
+
+import org.marc4j.MarcReader;
+import org.marc4j.MarcXmlReader;
+import org.marc4j.marc.ControlField;
+import org.marc4j.marc.DataField;
+import org.marc4j.marc.Record;
+import org.marc4j.marc.Subfield;
+
+import de.tuebingen.uni.ub.hc.Corpus.IxTheoCorpus;
+import de.tuebingen.uni.ub.hc.Corpus.IxTheoRecord;
+import de.tuebingen.uni.ub.hc.enums.IxTheoAnnotation;
+
+public class MARC4JProcessor {
+    private static InputStream in;
+    private static IxTheoCorpus corpus;
+
+    public static IxTheoCorpus getCorpus() {
+        return corpus;
+    }
+
+    public static void main(String args[]) throws Exception {
+        corpus = new IxTheoCorpus();
+        in = new FileInputStream("data/GesamtTiteldaten-post-pipeline-160612.xml");
+        // MarcReader reader = new MarcStreamReader(in);
+        processMARCRecords();
+
+    }
+
+    public static void processMARCRecords() throws FileNotFoundException {
+        MarcReader reader = new MarcXmlReader(in);
+        // FileOutputStream fw = new FileOutputStream(new
+        // File("data/gerCorpus.xml"));
+        // MarcWriter writer = new MarcXmlWriter(fw, "UTF8");
+        int countRecForTest = 0;
+        IxTheoRecord ixtRecord;
+        // iterate through all records in MARC corpus at pathname
+        while (reader.hasNext()) {
+            Record record = reader.next();
+            String data = "";
+            Subfield subfield;
+            String ppn = "0";
+            String author = "0";
+            String secAuthor = "0";
+            String lang = "0";
+            String title = "0";
+            String subtitle = "0";
+            String authorGND = "0";
+            String secAuthorGND = "0";
+            // get control field with tag 652 IXTHEOAnno
+            DataField currentField = (DataField) record.getVariableField("652");
+            // if record has IXTheo Anno add to our corpus
+            if (currentField != null) {
+                ppn = record.getControlNumber();
+                TreeSet<IxTheoAnnotation> ixTheoAnnoSet = new TreeSet<>();
+                // get Annotation(s)
+                data = currentField.getSubfieldsAsString("a");
+                if (!data.contains(":")) {
+                    ixTheoAnnoSet.add(IxTheoAnnotation.stringToIxTheo_Annotation(data));
+                } else {
+                    String[] annotations = data.split(":");
+                    for (int i = 0; i < annotations.length; i++) {
```

Comment:
I suspect a for-in loop might work here.  If yes, we always prefer for-in loops.